docker/templates/kaui/latest/setenv2.sh

```diff
@@ -17,6 +17,6 @@ fi
 if [ ! -z ${KAUI_CONFIG_DAO_PASSWORD+x} ]; then
   export KAUI_DB_PASSWORD=$KAUI_CONFIG_DAO_PASSWORD
 fi
-if [ ! -z ${KAUI_SECRET_KEY_BASE+x} ]; then
```

Comment:
👍 

This property is indeed different from the legacy properties above (e.g. `KAUI_DB_ADAPTER` is the old name for `KAUI_CONFIG_DAO_ADAPTER`). Here, if `KAUI_SECRET_KEY_BASE` isn't set, we want to set it. Maybe it's best to put it above the `Load legacy properties (backward compatibility)` comment above, so we don't get confused.

Quick test:

```
if [  -z ${KAUI_SECRET_KEY_BASE+x} ]; then
  export KAUI_SECRET_KEY_BASE=$(head -c 1024 /dev/urandom | base64 | tr -cd "[:upper:][:digit:]" | head -c 129)
fi
# Default value is set
echo $KAUI_SECRET_KEY_BASE
# 01G8SJN44WYARWB786LPSN3DEXCZ7PNBE4TS7O61DDMBS2LDLDW29EZU0D62RKR7XC468TD97UCFX2ZOEZAD08L8TEK85PR05SK0DO7IWSZ19X3PL14KYC93Q11CKTQLG

KAUI_SECRET_KEY_BASE=SOMETHING
if [  -z ${KAUI_SECRET_KEY_BASE+x} ]; then
  export KAUI_SECRET_KEY_BASE=$(head -c 1024 /dev/urandom | base64 | tr -cd "[:upper:][:digit:]" | head -c 129)
fi
# Specified value not overridden
echo $KAUI_SECRET_KEY_BASE
# SOMETHING
```

From https://www.gnu.org/software/bash/manual/html_node/Bash-Conditional-Expressions.html:

```
-z string
    True if the length of string is zero.
```